src/bindings/python/src/pyopenvino/frontend/extension.hpp

```diff
@@ -5,9 +5,34 @@
 #pragma once
 
 #include <pybind11/pybind11.h>
+#include <variant>
+#include "openvino/frontend/extension/conversion.hpp"
+#include "openvino/frontend/extension/op.hpp"
+#include "pyopenvino/graph/op.hpp"
 
 namespace py = pybind11;
 
+class PyFrontendOpExtension : public ov::frontend::ConversionExtensionBase {
+public:
+    explicit PyFrontendOpExtension(const std::string& op_type) 
+    : ov::frontend::ConversionExtensionBase(op_type) {}
+
+    PyFrontendOpExtension(const py::object& dtype, const std::string& op_type) : ConversionExtensionBase(op_type), py_handle_dtype{dtype} {
+        py::object py_issubclass = py::module::import("builtins").attr("issubclass");
+        if (!py_issubclass(dtype, py::type::of<PyOp>()).cast<bool>()) {
+            std::stringstream str;
+            str << "Unsupported data type : '" << dtype.attr("__name__").cast<std::string>() << "' is passed as an argument.";
+            OPENVINO_THROW(str.str());
+        }
+    }
+
+    std::variant<std::shared_ptr<ov::frontend::OpExtension<void>>, 
```

Comment:
`ov::frontend::OpExtension<void>`  is mapped by pybind to `OpExtension`. After this PR `ov::frontend::OpExtension<void>` it will be kept as `std::variant impl`. 

How is it currently used? If it will be std::variant, then to access C++ object sb will have to call `std::get<ov::frontend::OpExtension<void>>(impl)` and know the index or the type of variant.